src/Utilities/Conservation/Conservation.jl

```diff
@@ -0,0 +1,29 @@
+module Conservation
+
+using ..DGMethods
+using ..Mesh.Grids
+using ..MPIStateArrays
+
+using MPI
+
+function compute(mpicomm::MPI.Comm, dg::DGModel, Q::MPIStateArray)
+    vgeo = dg.grid.vgeo
+
+    M = view(dg.grid.vgeo, :, [Grids._M], :)
+
+    Σρ = MPI.Reduce(sum(Q.ρ .* M), +, 0, mpicomm)
+    Σρe = MPI.Reduce(sum(Q.ρe .* M), +, 0, mpicomm)
```

Comment:
these lines only work for an `AtmosModel`, they will fail on any other balance law. I think it would be better practice to pass in the symbol of which state variable you want to compute as an argument to this function and then configure the Atmos specific part in the Atmos specific diagnostics

Reply:
Would be better to use `weightedsum` https://github.com/CliMA/ClimateMachine.jl/blob/64fbdf3556bd2ef6b8958670308dde7f96416cf3/src/Arrays/MPIStateArrays.jl#L594-L622

With this and `varsindices` https://github.com/CliMA/ClimateMachine.jl/blob/64fbdf3556bd2ef6b8958670308dde7f96416cf3/src/Utilities/VariableTemplates/VariableTemplates.jl#L58-L85

should generalize this fairly easily.

Reply:
Ah, yeah. I was thinking of putting that off, but I'll fix it in this PR.

Reply:
Fixed this, can you guys take a look please?

Reply:
Looks fine to me.